### PR TITLE
Cleanups

### PR DESCRIPTION
**Pull request by lucasdemarchi**
This is a collection of cleanups I'm moving from the #3544 PR and sending as a separate PR - this allows to get at least part of the improvements sooner, reducing the size of that PR.

### CODE REVIEW

File: libraries/AP_Baro/AP_Baro_MS5611.h [hunk 1]
```diff
@@ -88,19 +88,19 @@ class AP_Baro_MS56XX : public AP_Baro_Backend
     AP_SerialBus *_serial;
 
     /* Asynchronous state: */
-    volatile bool            _updated;
+    volatile bool            _updated = false;
```
Comment (tridge) [inline libraries/AP_Baro/AP_Baro_MS5611.h]: adding zero initialisers like this increases the size of the PX4 firmware without actually achieving anything. I started to do that myself a while ago then noticed the sharp increase in PX4 firmware size. Without our new/malloc wrapper guaranteeing zero init I think we should leave these out

File: libraries/AP_HAL/utility/functor.h [hunk 2]
```diff
@@ -15,10 +15,9 @@
  * You should have received a copy of the GNU General Public License along
  * with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
-
 #pragma once
-#ifndef __FUNCTOR_H__
-#define __FUNCTOR_H__
+
+#include <type_traits>
```
Comment (tridge) [inline libraries/AP_HAL/utility/functor.h]: this breaks the qurt build, with:
/home/tridge/project/UAV/APM/libraries/AP_HAL/utility/functor.h:20:10: fatal error: 'type_traits' file not found
looks like we don't get type_trains on clang with -std=gnu++11

Comment (lucasdemarchi) [inline libraries/AP_HAL/utility/functor.h]: clang certainly has it.... maybe not the one used in QURT.

So, it's not time yet to do this change. I'll remove this patch.

File: libraries/AP_Baro/AP_Baro_MS5611.h (hunk 1 above)
Comment (lucasdemarchi) [inline libraries/AP_Baro/AP_Baro_MS5611.h]: Actually removing from the constructor's initializer list and putting it here wouldn't increase the size. However we can just remove them and reduce the size a little bit.

**Comment by lucasdemarchi** [comment]
@tridge I removed the commit for type_traits and changed the MS5611 one to remove initialization to 0 from the constructor.

**Comment by guludo** [comment]
I think the only real issue I see is the one related to the SPI device name (https://github.com/diydrones/ardupilot/pull/3564#commitcomment-15799409). Other than that, looks good to me.

**Comment by lucasdemarchi** [comment]
@guludo thanks. I fixed those and applied.

**Pull request closed without merging**